lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -702,6 +715,99 @@ export abstract class ClientApplication {
+        }
+
+        const popupName = PopupUtils.generateLogoutPopupName(this.config.auth.clientId, validLogoutRequest);
+        let popup;
+
+        // asyncPopups flag is true. Acquires token without first opening popup. Popup will be opened later asynchronously.
+        if (this.config.system.asyncPopups) {
+            this.logger.verbose("asyncPopups set to true");
+        } else {
+            // asyncPopups flag is set to false. Opens popup before logging out.
+            this.logger.verbose("asyncPopup set to false, opening popup");
+            popup = PopupUtils.openSizedPopup("about:blank", popupName);
+        }
+
+        const authority = logoutRequest && logoutRequest.authority;
+        const redirectMainWindowTo = logoutRequest && logoutRequest.redirectMainWindowTo;
+        return this.logoutPopupAsync(validLogoutRequest, popupName, authority, popup, redirectMainWindowTo);
+    }
+
+    /**
+     * 
+     * @param request 
+     * @param popupName 
+     * @param requestAuthority
+     * @param popup 
+     */
+    private async logoutPopupAsync(validRequest: CommonEndSessionRequest, popupName: string, requestAuthority?: string, popup?: Window|null, redirectMainWindowTo?: string): Promise<void> {
+        this.logger.verbose("logoutPopupAsync called");
+        this.emitEvent(EventType.LOGOUT_START, InteractionType.Popup, validRequest);
+        
+        const serverTelemetryManager = this.initializeServerTelemetryManager(ApiId.logoutPopup, validRequest.correlationId);
+        
+        try {
+            this.browserStorage.setTemporaryCache(TemporaryCacheKeys.INTERACTION_STATUS_KEY, BrowserConstants.INTERACTION_IN_PROGRESS_VALUE, true);
+            // Initialize the client
+            const authClient = await this.createAuthCodeClient(serverTelemetryManager, requestAuthority);
+
+            // create logout string and navigate user window to logout. Auth module will clear cache.
+            const logoutUri: string = authClient.getLogoutUri(validRequest);
+            if (!validRequest.account || AccountEntity.accountInfoIsEqual(validRequest.account, this.getActiveAccount())) {
+                this.logger.verbose("Setting active account to null");
+                this.setActiveAccount(null);
+            }
+
+            this.emitEvent(EventType.LOGOUT_SUCCESS, InteractionType.Popup, validRequest);
+
+            const popupUtils = new PopupUtils(this.browserStorage, this.logger);
+            // Open the popup window to requestUrl.
+            const popupWindow = popupUtils.openPopup(logoutUri, popupName, popup);
+            this.emitEvent(EventType.POPUP_OPENED, InteractionType.Popup, {popupWindow}, null);
+
+            try {
+                // Don't care if this throws an error (User Cancelled)
+                await popupUtils.monitorPopupForSameOrigin(popupWindow);
+            } catch {}
+
+            popupUtils.cleanPopup(popupWindow);
+
+            if (redirectMainWindowTo) {
```

Comment:
Can you add a log message for the if and else cases for this block?

Reply:
Done